src/kOS.Safe/Utilities/Debug.cs

```diff
@@ -63,5 +64,33 @@ public static List<string> GetPendingNags()
         }
 
         public static ObjectIDGenerator IDGenerator { get; set; }
+
+        /// <summary>
+        /// This is copied almost verbatim from ProgramContext,
+        /// It's here to help debug.
+        /// </summary>
+        public static string GetCodeFragment(List<Opcode> codes)
+        {
+            var codeFragment = new List<string>();
+            
+            const string FORMAT_STR = "{0,-20} {1,4}:{2,-3} {3:0000} {4} {5} {6} {7}";
+            codeFragment.Add(string.Format(FORMAT_STR, "File", "Line", "Col", "IP  ", "Label  ", "opcode", "operand", "Destination" ));
+            codeFragment.Add(string.Format(FORMAT_STR, "----", "----", "---", "----", "-------", "---------------------", "", "" ));
+
+            for (int index = 0; index < codes.Count; index++)
+            {
+                codeFragment.Add(string.Format(FORMAT_STR,
+                                               codes[index].SourceName ?? "null",
+                                               codes[index].SourceLine,
+                                               codes[index].SourceColumn ,
+                                               index,
+                                               codes[index].Label ?? "null",
+                                               codes[index] ?? new OpcodeBogus(),
+                                               "DEST: " + (codes[index].DestinationLabel ?? "null" ),
+                                               "" ) );
+            }
+
+            return codeFragment.Aggregate(string.Empty, (current, s) => current + (s + "\n"));
+        }
```

Comment:
I moved the above function out of kOS Utilities into kOS.Safe Utilities so I could call it from places in the compiler during my checking.  It seemed like a good move so I left it there in my PR.